### ISSUE TITLE
Animator bug

### ISSUE THREAD

**Issue by SnaiperoG3D** [FirstGearGames/FishNet#179]
In short: na takes wrong animation from animator state and send to clients, because animator state not changed yet after Play call. i think it happens only with complex logic and related to Unity Animator what not update itself yet in frame when Play called.

Discord msgs: 

i have a feeling that na getting animator state which not changed yet, i had this issue when wrote a code controlling weapon enabling and disabling, i found that it need 1-2 frames to wait before animator state actually will update itself, or maybe i just miss something, but i will try to make a delay in na, or just print the value, its only one thought that i have for now, even if its the reason, its not possible that only i facing it. damn its blowing my mind 😩 

so ive putted a debug into na part, where it takes current animator state info, and it is
animation name of GetCurrentAnimatorClipInfo is previos one, what was before call Play

also strange thing, na using ticks for calculations, and reducing ticks count should fix it, if it so, cuz i guess it should pass enough time between ticks for animator to update it state, but it not help. 😩

created a test scene in empty project with same debugs in na, all fine, so something influencing on animator. i guess its a unity bug, found this thread (https://forum.unity.com/threads/how-to-make-instant-animation-change-with-no-frames-in-between.1204219/), dont know to load animator to catch issue yet with coroutines

so, ive did a temp not elegant but working solution as i did before for my wep controll, i did a wrap for na.play in my script, which basically run animation on Animator, wait one frame and then call na.play. after this frame wait everything works fine

Code snippet: https://pastebin.com/gkvT6ZSj

**Comment by SnaiperoG3D** [comment]
Also, same thing happens with layer weights, i do in my code changing layer weights and play anim in NA, so i added into cor changing layers weight too after one frame wait, cuz na takes previos anim and layers too, basically it take all info which not changed yet

**Comment by FirstGearGames** [comment]
Do you only have to wait 1 frame?

**Comment by SnaiperoG3D** [comment]
yeah, i have one theory about unity animator, why it can do this issue, ill try to check it later

**Comment by FirstGearGames** [comment]
Follow-up?

**Comment by FirstGearGames** [comment]
Closing due to inactivity.

**Comment by SnaiperoG3D** [comment]
ive tried to check my theory but no luck, i guess there is something wrong with unity animator, something causing unity to process animation change only in the next frame, or something like that, i tried to load animator by layers and clips in empty project, but no issue reproducing
